internal/kubernetes/operator/dbusers/dbusers_test.go

```diff
@@ -0,0 +1,356 @@
+				},
+			},
+		}
+
+		expectedLabels := []common.LabelSpec{
+			{
+				Key:   "TestKey",
+				Value: "TestValue",
+			},
+		}
+
+		if got := convertUserLabels(atlasUser); !reflect.DeepEqual(got, expectedLabels) {
+			t.Errorf("convertUserLabels() = %v, want %v", got, expectedLabels)
+		}
+	})
+}
+
+func Test_convertUserRoles(t *testing.T) {
+	t.Run("Can convert user labels from Atlas to the Operator format", func(t *testing.T) {
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Roles: []mongodbatlas.Role{
+				{
+					RoleName:       "TestRole",
+					DatabaseName:   "TestDB",
+					CollectionName: "TestCollection",
+				},
+			},
+		}
+
+		expectedRoles := []atlasV1.RoleSpec{
+			{
+				RoleName:       "TestRole",
+				DatabaseName:   "TestDB",
+				CollectionName: "TestCollection",
+			},
+		}
+		if got := convertUserRoles(atlasUser); !reflect.DeepEqual(got, expectedRoles) {
+			t.Errorf("convertUserRoles() = %v, want %v", got, expectedRoles)
+		}
+	})
+}
+
+func Test_buildUserSecret(t *testing.T) {
+	t.Run("Can build user secret WITHOUT credentials", func(t *testing.T) {
+		projectName := "TestProject-1"
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Password: "TestPassword",
+			Username: "TestName",
+		}
+
+		expectedSecret := &corev1.Secret{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Secret",
+				APIVersion: "v1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", projectName, atlasUser.Username),
+				Namespace: "TestNamespace",
+				Labels: map[string]string{
+					secrets.TypeLabelKey: secrets.CredLabelVal,
+				},
+			},
+			Data: map[string][]byte{
+				PasswordField: []byte(""),
+			},
+		}
+		if got := buildUserSecret(projectName, atlasUser, "TestNamespace", false); !reflect.DeepEqual(got, expectedSecret) {
+			t.Errorf("buildUserSecret(); \r\n got:%v;s\r\n want:%v", got, expectedSecret)
+		}
+	})
+
+	t.Run("Can build user secret WITH credentials", func(t *testing.T) {
+		projectName := "TestProject-2"
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Password: "TestPassword",
+			Username: "TestName",
+		}
+
+		expectedSecret := &corev1.Secret{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Secret",
+				APIVersion: "v1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", projectName, atlasUser.Username),
+				Namespace: "TestNamespace",
+				Labels: map[string]string{
+					secrets.TypeLabelKey: secrets.CredLabelVal,
+				},
+			},
+			Data: map[string][]byte{
+				PasswordField: []byte(atlasUser.Password),
+			},
+		}
+		if got := buildUserSecret(projectName, atlasUser, "TestNamespace", true); !reflect.DeepEqual(got, expectedSecret) {
+			t.Errorf("buildUserSecret(); \r\n got:%v;s\r\n want:%v", got, expectedSecret)
+		}
+	})
+}
+
+type MockUserStore struct {
+	Users map[string][]mongodbatlas.DatabaseUser
+}
+
+func (m *MockUserStore) DatabaseUsers(groupID string, _ *mongodbatlas.ListOptions) ([]mongodbatlas.DatabaseUser, error) {
+	return m.Users[groupID], nil
+}
```

Comment:
any reason you are not using https://github.com/golang/mock

Reply:
This comment is not addressed.